### ISSUE TITLE
Series time_name argument is ignored

### ISSUE THREAD

**Issue by chrishavlin** [LinkedEarth/Pyleoclim_util#469]
**Describe the bug**
New to pyleoclim, so this may be me misunderstanding things a bit, but the optional `time_name` argument to the `Series` object seems to be ignored. 

**To Reproduce**
```python
import pyleoclim as pyleo
import numpy as np

omega = 0.5
t = np.linspace(0, 20, 1000)
b = np.cos(2 * np.pi * omega * t + - np.pi / 4)

b_series = pyleo.Series(time=t, value=b,
                      time_name='my age', time_unit='ka',
                      value_name='b')
b_series.plot()
```
![Screenshot from 2023-09-19 09-48-40](https://github.com/LinkedEarth/Pyleoclim_util/assets/22038879/4602dfa3-f969-4369-b9b7-c3ac70c5677d)

**Expected behavior**
I'd expect `'my age'` on the x axis. 

**Desktop (please complete the following information):**
 - OS: ubuntu 

**Additional context**
pyleoclim version 0.13.0

Looking at how `Series` gets initialized, it seems the time units disambiguation is setting the `time_name` regardless of whether or not I specified `time_name`, so maybe this is not a bug and is the intended behavior, but if that's the case it'd be helpful if the docstring entry for `time_name` could explain conditions where `time_name` would be used.

**Comment by CommonClimate** [comment]
Hi Chris,
yes, you looked in the right place, and indeed, this line takes precedence here:
 `time_name, time_unit = tsbase.disambiguate_time_metadata(time_unit)`. A specified `time_name` is only used if it belongs to authorized entries: 
 ```

```
I do agree with you that the docstring could be clearer on this point (my bad), so I will modify it accordingly, hoping nothing breaks.  Is there a particular nomenclature you are trying to use outside of "Time" or "Age"?

**Comment by CommonClimate** [comment]
As expected, some tests failed, which were meant to catch real-world contradictions. At this point I am reluctant to allow custom time_names beyond Time and Age, unless you have a strong rationale for another choice.

**Comment by chrishavlin** [comment]
Thanks for the extra context, @CommonClimate ! I think my misunderstanding came from not realizing that `time_name` is more than just an axis label here. Having a note somewhere in the docs or in the docstring to that effect would be helpful (maybe it's already there somewhere but I missed it).

In terms of what nomenclature I'm trying to use... I was actually looking into using pyleoclim's wavelet anlalysis for applications to more general timeseries (some timeseries analysis of experimental rock deformation data, maybe some seismic applications). So I definitely wouldn't recommend making a change for my sake here :)  I've found a couple other packages for wavelet analysis, but pyleoclim is more actively maintained than others, so I started with it. And after submitting this issue I spent some time looking through the API in more detail and I actually think for my purposes I'll try out using the `utils.wavelet` functions directly. 

So feel free to close this issue at any point.

**Comment by CommonClimate** [comment]
Hi @chrishavlin, not a problem: this is something we need to straighten out anyway, and it's now on the agenda for our development meeting this Friday. 

Happy to hear that you are finding Pyleoclim's wavelet capabilities useful! If you are unafraid to go down the utils level functions, that is definitely a good path. Otherwise the APIs can always be hacked; for instance, since everything is Matplotlib-based, it's easy to overwrite an xlabel or ylabel argument if you did not get exactly what you wanted on the first go around. 

We strive for a balance between ease of use for our main constituency (paleo folk) and flexibility for everyone else, but in this case we appear to have hit a point of tension. I'll let you know what gets decided Friday. Thanks for your patience!

**Comment by CommonClimate** [comment]
Proposed Solution:
- introduce an `auto_time_params` parameter in `Series.___init___` , with possible values 'on' or 'off' (default). 
- If 'on', then trigger the guesswork we currently have (which is mainly aimed for `common_time()`). 
- If 'off', then leave `time_name` and `time_unit` as they are. The docstring should warn users that this might break various things, but with great power comes great responsibility. 
- @alexkjames agreed to prototype this

**Comment by CommonClimate** [comment]
resolved by #472

**Comment by CommonClimate** [comment]
Hi @chrishavlin , sorry for the time it took to address this; @alexkjames has implemented a fix, and setting `auto_time_params = False` upon creating your Series should do the trick. Please give it a try and feel free to re-open the issue if it persists!
